pkg/sql/logictest/logic.go

```diff
@@ -3849,6 +3876,7 @@ func (t *logicTest) formatValues(vals []string, valsPerLine int) []string {
 }
 
 func (t *logicTest) success(file string) {
+	t.retry = false
```

Comment:
What if success isn't called?   I wonder if this should be set to false at the top of processSubtest to be safe.